### PR TITLE
Add error handling to Optimize and Translate

### PR DESCRIPTION
**Pull request by jkhales**
Added basic validation and error handling to the two main functions in transformations.go, `Optimize` and `Translate` as requested in:

https://github.com/TimothyStiles/poly/issues/134

It could be nice to add some typing/validation to amino acid and sequence strings that do a regex check.

### CODE REVIEW

**Comment by jkhales** [comment]
Also, is `poly` versioned in any sense? This would definitely be a "breaking" change.

**Comment by TimothyStiles** [comment]
Yeah we are @jkhales. If you look at releases on the right side of the project main you can see what we're on. I usually put BREAKING CHANGES on release notes and that seems to do the trick for now.

**Comment by jkhales** [comment]
> Yeah we are @jkhales. If you look at releases on the right side of the project main you can see what we're on. I usually put BREAKING CHANGES on release notes and that seems to do the trick for now.

Nice, so there is no changelog or anything I need to edit? Or is there something else to do for this PR?

**Comment by TimothyStiles** [comment]
@jkhales is there any way to test error handling too? Would be good to have in tests other than just assigning to `_` and ignoring.

**Comment by rkrishnasanka** [comment]
> Yeah we are @jkhales. If you look at releases on the right side of the project main you can see what we're on. I usually put BREAKING CHANGES on release notes and that seems to do the trick for now.

There needs to be a version bump if its breaking changes (minor at least)

**Comment by jkhales** [comment]
> @jkhales is there any way to test error handling too? Would be good to have in tests other than just assigning to `_` and ignoring.

Did you see these tests that I added?

https://github.com/TimothyStiles/poly/pull/151/files#diff-bdccce56349e8b0b47fc7d38f6449396ae832f9491993497189daa63544f5935R32

https://github.com/TimothyStiles/poly/pull/151/files#diff-bdccce56349e8b0b47fc7d38f6449396ae832f9491993497189daa63544f5935R82

File: transformations.go [hunk 1]
```diff
@@ -55,6 +55,10 @@ Tim
 
 ******************************************************************************/
 
+var errEmtpyCodonTable error = errors.New("empty codon table")
+var errEmtpyAminoAcidString error = errors.New("empty amino acid string")
+var errEmtpySequenceString error = errors.New("empty sequence string")
```
Comment (TimothyStiles) [inline transformations.go:60]: removed as global variable since it's only being used once.

**Pull request merged**